From 2c57fd6f7e03fd95459c8274e97406526709c4fd Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Fri, 23 Aug 2024 12:22:06 +0200
Subject: [PATCH 7/9] On error in sr_shmsub_listen_thread(), exit process
Organization: Wires

If processing callback events in, e.g., sysrepo-plugind, make sure to
log the error and exit(1) the entire process so the system can decide
to handle the problem.  For example, restart all dependent services.

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/shm_sub.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/shm_sub.c b/src/shm_sub.c
index 3fe8c8e6..2ab31ed3 100644
--- a/src/shm_sub.c
+++ b/src/shm_sub.c
@@ -5035,5 +5035,7 @@ error:
     /* free our own resources */
     ATOMIC_STORE_RELAXED(subscr->thread_running, 0);
     pthread_detach(pthread_self());
+    exit(1);
+
     return NULL;
 }
